### PR TITLE
feat(model): Resume trainable lightning fabric pipeline

### PR DESCRIPTION
**Pull request by MoosaTae**
## Why this PR
because training fabric pipeline not available for now

## Changes
- Resume Training is possible
- change .to("cuda") to with fabric.device for adamw fuse mode
- use fabric.print instead print
- solve wrong parameters name
- rename llama to 100M size params

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/model/openthaigpt_pretraining_model/lightning/utils.py [hunk 1]
```diff
@@ -123,28 +154,40 @@ def val_step(self):
...
-            if not is_accumulating:
-                self.opt.step()
-                self.opt.zero_grad()
-
-        val_loss = self.val_step()
-        print(f"loss_val: {val_loss.item():.3f}")
+        for epoch in range(self.epochs):
+            progress_bar = tqdm(self.dataloader, disable=(self.fabric.global_rank != 0))
+            for i, batch in enumerate(progress_bar):
+                if i < self.start_steps and epoch == 0:
```
[the remark this answers is not in the document]
Comment (MoosaTae) [reply src/model/openthaigpt_pretraining_model/lightning/utils.py:165]: Should I do resume trainging from epoch that saved?
because it just can reset to 0?

[the remark this answers is not in the document]
Comment (boat1603) [reply src/model/openthaigpt_pretraining_model/lightning/utils.py:165]: You have to resuming from latest epoch too.

File: src/model/openthaigpt_pretraining_model/lightning/utils.py [hunk 2]
```diff
@@ -123,28 +154,40 @@ def val_step(self):
...
+                    if (i + 1) % self.eval_steps == 0:
+                        (val_loss, perplexity) = self.val_step()
+                        self.fabric.print(
+                            f"val_loss: {val_loss.item():.3f}, "
+                            f"perplexity: {perplexity:.3f})"
+                        )
+
+                if (i + 1) % self.save_steps == 0:
+                    self.fabric.print(f"Saving weights : {self.save_paths}")
+                    self.save_checkpoint(i + 1, epoch)
```
[the remark this answers is not in the document]
Comment (MoosaTae) [reply src/model/openthaigpt_pretraining_model/lightning/utils.py:191]: Is it will skip until it reach begin step?

File: src/model/configuration_example/training/training_v1.yaml [hunk 3]
```diff
@@ -11,4 +11,11 @@ batch_size: 2
...
 num_shards: 1024
-num_workers: 2
\ No newline at end of file
+num_workers: 2
+epochs: 1
+start_epochs: 0
+start_steps: 0
+save_steps: 10000
+eval_steps: 10000
+save_paths: path/to/weight
```
Comment (boat1603) [inline src/model/configuration_example/training/training_v1.yaml:20]: I think just ./weight is OK.

**Review by boat1603 — APPROVED** [review]

File: src/model/configuration_example/training/training_v1.yaml (hunk 3 above)
Comment (MoosaTae) [reply src/model/configuration_example/training/training_v1.yaml]: done

**Review by new5558 — COMMENTED** [review]
From what I understand from the code. This PR only save and load model weights not training info like scheduler, optimizer etc.


Please modify your code to incorporate that

File: src/model/configuration_example/training/training_v1.yaml [hunk 4]
```diff
@@ -11,4 +11,11 @@ batch_size: 2
 grad: 4
 max_tokens: 2048
 num_shards: 1024
-num_workers: 2
\ No newline at end of file
+num_workers: 2
+epochs: 1
+start_epochs: 0
```
Comment (new5558) [inline src/model/configuration_example/training/training_v1.yaml:16]: What the meaning of start steps and start epoch. Can you explain in the yaml comments?

File: src/model/configuration_example/training/training_v1.yaml [hunk 5]
```diff
@@ -11,4 +11,11 @@ batch_size: 2
...
+start_steps: 0
+save_steps: 10000
+eval_steps: 10000
+save_paths: ./weight
+weight_path: False
```
Comment (new5558) [inline src/model/configuration_example/training/training_v1.yaml:21]: What is the different between weight path and save path. Can you write in comments?

File: src/model/configuration_example/training/training_v1.yaml [hunk 6]
```diff
@@ -11,4 +11,11 @@ batch_size: 2
 grad: 4
 max_tokens: 2048
 num_shards: 1024
-num_workers: 2
\ No newline at end of file
+num_workers: 2
+epochs: 1
```
Comment (new5558) [inline src/model/configuration_example/training/training_v1.yaml:15]: What if I want to just train for 100000 steps not epoch?

File: src/model/configuration_example/training/training_v1.yaml (hunk 5 above)
Comment (MoosaTae) [reply src/model/configuration_example/training/training_v1.yaml:21]: I forgot to remove

**Review by new5558 — APPROVED** [review]
